.husky/pre-commit

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo 'Checking types, formatting, linting files before committing...'
+
+# Check prettier standards
+npm run check-format ||
+(
+  echo 'Prettier check failed! Run "npm run format", add changes and try commit again.'
+  false;
+)
+
+# Check ESLint standards
+npm run check-lint ||
+(
+  echo 'ESLint check failed! Make the required changes listed above, add changes and try to commit again.'
+  false;
+)
+
+# Check tsconfig standards
+npm run check-types ||
+(
+  echo 'Make changes required above and try again.'
+  false;
+)
+
+# Everything went well so far, lets build
+npm run build ||
+(
+  echo 'Next build failed. See the errors above.'
+  false;
+)
```

Comment:
I think we don't need to build it on every commit cause that can slow things down significantly.

Reply:
That's true, I'll remove it.